staticroute.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package gomaasapi
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/schema"
+	"github.com/juju/version"
+)
+
+type staticRoute struct {
+	// Add the controller in when we need to do things with the staticRoute.
+	// controller Controller
+
+	resourceURI string
+
+	id          int
+	source      *subnet
+	destination *subnet
+	gatewayIP   string
+	metric      int
+}
+
+// Id implements StaticRoute.
+func (s *staticRoute) ID() int {
+	return s.id
+}
+
+// Source implements StaticRoute.
+func (s *staticRoute) Source() Subnet {
+	return s.source
+}
+
+// Destination implements StaticRoute.
+func (s *staticRoute) Destination() Subnet {
+	return s.destination
+}
+
+// GatewayIP implements StaticRoute.
+func (s *staticRoute) GatewayIP() string {
+	return s.gatewayIP
+}
+
+// Metric implements StaticRoute.
+func (s *staticRoute) Metric() int {
+	return s.metric
+}
+
+func readStaticRoutes(controllerVersion version.Number, source interface{}) ([]*staticRoute, error) {
+	checker := schema.List(schema.StringMap(schema.Any()))
+	coerced, err := checker.Coerce(source, nil)
+	if err != nil {
+		return nil, errors.Annotatef(err, "static-route base schema check failed")
```

Comment:
This is checking that we have a list of objects (eg [{}, {}, {}], which is supposed to be true regardless of what version of the API we are talking to.

I think this was a deliberate choice from the people implementing the maas parser that you break apart parsing a list of objects into a top level 'parse it into a list' and then intermediate 'parse it into version-specific objects'.